persistence/internal/providertest/offsetstore/repository.go

```diff
@@ -0,0 +1,41 @@
+package offsetstore
+
+import (
+	"context"
+
+	"github.com/dogmatiq/infix/persistence"
+	"github.com/dogmatiq/infix/persistence/internal/providertest/common"
+	"github.com/dogmatiq/infix/persistence/subsystem/offsetstore"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+// DeclareRepositoryTests declares a functional test-suite for a specific
+// offsetstore.Repository implementation.
+func DeclareRepositoryTests(tc *common.TestContext) {
+	ginkgo.Describe("type offsetstore.Repository", func() {
+		var (
+			dataStore  persistence.DataStore
+			repository offsetstore.Repository
+			tearDown   func()
+		)
+
+		ginkgo.BeforeEach(func() {
+			dataStore, tearDown = tc.SetupDataStore()
+			repository = dataStore.OffsetStoreRepository()
+
+		})
+
+		ginkgo.AfterEach(func() {
+			tearDown()
+		})
+
+		ginkgo.Describe("func LoadOffset()", func() {
+			ginkgo.It("loads the initial offset as zero", func() {
+				o, err := repository.LoadOffset(context.Background(), "<app-key>")
```

Comment:
I would suggest using a different string here, because all of the data-store that gets setup is for the app known as `<app-key>`. This way we make sure we're not swapping these values inadvertently.

The "source application" is existing terminology I have meaning the application that produced a message. In this case we're getting messages from the stream of their source application.

```suggestion
				o, err := repository.LoadOffset(context.Background(), "<source-app-key>")
```